message_processor.go

```diff
@@ -27,11 +31,53 @@ func (mg messageProcessor) process(lf *logPair) {
 	consumeLog(lf)
 }
 
+type tmpBuffer struct {
```

Comment:
I would suggest to refactor out `tmpBuffer` to a new golang file and change the name of `tmpBuffer` since it is a little bit confusing. Please make sure there are necessary public APIs for better unit testing.

Reply:
See above.